mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts

```diff
@@ -15,9 +17,50 @@ export class WayPreviewDAL {
    */
   public static async getWaysPreview(): Promise<WayPreview[]> {
     const waysDTO = await WayService.getWaysDTO();
-    const waysUuids = waysDTO.map((item) => item.uuid);
 
-    const waysPreview = await Promise.all(waysUuids.map(WayPreviewDAL.getWayPreview));
+    const allNeededUsersUuids = new Set(waysDTO.flatMap(wayDTO => [
+      wayDTO.ownerUuid,
+      ...wayDTO.mentorUuids,
+      ...wayDTO.mentorRequestUuids,
+    ]));
+
+    const allNeededGoalsUuids = new Set(waysDTO.flatMap(wayDTO => [wayDTO.goalUuid]));
```

Comment:
why cant we use const allNeededGoalsUuids = new Set(waysDTO.map(wayDTO => wayDTO.goalUuid));

Reply:
by the way, goals are unique for each way, why do you need to create a Set?

Reply:
Fixed